When using StartScan with timeout ctx: panic nil pointer dereference

### Expected Behavior
When calling pkg/scanner/scanner.go:51:StartScan  with timeout context, the Scanner should exit gracefully.

### Actual Behavior
```go
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x1180707]

goroutine 53 [running]:
github.com/Checkmarx/kics/pkg/engine.(*Inspector).Inspect(0xc0008b0bd0, {0x2cfd9b8?, 0xc0008d2060}, {0x2712ca6, 0x7}, {0x0?, 0x0, 0x0}, {0xc00061d0f0, 0x1, ...}, ...)
	C:/dev/go/kics/pkg/engine/inspector.go:226 +0x4e7
github.com/Checkmarx/kics/pkg/kics.(*Service).StartScan(0xc000c99d50, {0x2cfd9b8, 0xc0008d2060}, {0x2712ca6, 0x7}, 0x0?, 0xc0018fc570, 0x0?)
	C:/dev/go/kics/pkg/kics/service.go:101 +0x245
created by github.com/Checkmarx/kics/pkg/scanner.StartScan
	C:/dev/go/kics/pkg/scanner/scanner.go:67 +0x18e
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x1180707]

goroutine 50 [running]:
github.com/Checkmarx/kics/pkg/engine.(*Inspector).Inspect(0xc0008b0bd0, {0x2cfd9b8?, 0xc0008d2060}, {0x2712ca6, 0x7}, {0x0?, 0x0, 0x0}, {0xc00061d0f0, 0x1, ...}, ...)
	C:/dev/go/kics/pkg/engine/inspector.go:226 +0x4e7
github.com/Checkmarx/kics/pkg/kics.(*Service).StartScan(0xc000c99b90, {0x2cfd9b8, 0xc0008d2060}, {0x2712ca6, 0x7}, 0x43475f3832315f41?, 0xc0018fc570, 0x5f534c5422202c22?)
	C:/dev/go/kics/pkg/kics/service.go:101 +0x245
created by github.com/Checkmarx/kics/pkg/scanner.StartScan
	C:/dev/go/kics/pkg/scanner/scanner.go:67 +0x18e
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x1180707]

goroutine 52 [running]:
github.com/Checkmarx/kics/pkg/engine.(*Inspector).Inspect(0xc0008b0bd0, {0x2cfd9b8?, 0xc0008d2060}, {0x2712ca6, 0x7}, {0x0?, 0x0, 0x0}, {0xc00061d0f0, 0x1, ...}, ...)
	C:/dev/go/kics/pkg/engine/inspector.go:226 +0x4e7
github.com/Checkmarx/kics/pkg/kics.(*Service).StartScan(0xc000c99ce0, {0x2cfd9b8, 0xc0008d2060}, {0x2712ca6, 0x7}, 0x0?, 0xc0018fc570, 0x0?)
	C:/dev/go/kics/pkg/kics/service.go:101 +0x245
created by github.com/Checkmarx/kics/pkg/scanner.StartScan
	C:/dev/go/kics/pkg/scanner/scanner.go:67 +0x18e


Process finished with the exit code 1
```

### Steps to Reproduce the Problem
I ran the test `TestScanner_StartScan` with `context.WithTimeout` to simulate a `ctx.Done` event:
```diff
--- a/pkg/scanner/scanner_test.go
+++ b/pkg/scanner/scanner_test.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"path/filepath"
 	"testing"
+	"time"
 
 	"github.com/Checkmarx/kics/assets"
 	"github.com/Checkmarx/kics/internal/storage"
@@ -56,11 +57,14 @@ func TestScanner_StartScan(t *testing.T) {
 		},
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			services, store, err := createServices(tt.fields.types, tt.fields.cloudProviders)
 			require.NoError(t, err)
-			err = StartScan(context.Background(), tt.args.scanID, progress.PbBuilder{}, services)
+			err = StartScan(ctx, tt.args.scanID, progress.PbBuilder{}, services)
 			require.NoError(t, err)
 			require.NotEmpty(t, &store)
 		})


```

### Root cause 
```go
//  pkg/engine/inspector.go:226
                queryOpa, err := c.QueryLoader.LoadQuery(ctx, &queries[i])
		if err != nil {
			continue
		}

		log.Debug().Msgf("Starting to run query %s", queryMeta.Query)
		queryStartTime := time.Now()

		query := &PreparedQuery{
			OpaQuery: *queryOpa, // <--nil pointer dereference !
			Metadata: queryMeta,
		}
```
```go
// pkg/engine/inspector.go:474
func (q QueryLoader) LoadQuery(ctx context.Context, query *model.QueryMetadata) (*rego.PreparedEvalQuery, error) {
	opaQuery := rego.PreparedEvalQuery{}

	platformGeneralQuery, ok := q.platformLibraries[query.Platform]
	if !ok {
		return nil, errors.New("failed to get platform library")
	}

	select {
	case <-ctx.Done():
		return nil, nil
```



As you can see `LoadQuery` returns `nil, nil` when ctx.Done fires, so the solution could be:
1. Return error in `LoadQuery` of context deadline reached.
2. Add `if queryOpa == nil {continue}` after the error check.
<br>
Let me know which solution sound good for you, and I'll create a PR for it :)

### Specifications
-   Version:  commit 174648e0e42dd16204bd5cd04c5394e491403a5a (HEAD -> master, origin/master, origin/HEAD)
-   Platform: Windows

--- a/pkg/engine/inspector.go
+++ b/pkg/engine/inspector.go
@@ -481,7 +481,7 @@ func (q QueryLoader) LoadQuery(ctx context.Context, query *model.QueryMetadata) 
 
 	select {
 	case <-ctx.Done():
-		return nil, nil
+		return nil, ctx.Err()
 	default:
 		mergedInputData, err := source.MergeInputData(platformGeneralQuery.LibraryInputData, query.InputData)
 		if err != nil {

--- a/pkg/scanner/scanner_test.go
+++ b/pkg/scanner/scanner_test.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"path/filepath"
 	"testing"
+	"time"
 
 	"github.com/Checkmarx/kics/assets"
 	"github.com/Checkmarx/kics/internal/storage"
@@ -29,6 +30,11 @@ var (
 	sourcePath = []string{filepath.FromSlash("../../assets/queries")}
 )
 
+type testContext struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
 func TestScanner_StartScan(t *testing.T) {
 	type args struct {
 		scanID     string
@@ -38,13 +44,16 @@ func TestScanner_StartScan(t *testing.T) {
 		types          []string
 		cloudProviders []string
 	}
+
+	ctx := context.Background()
 	tests := []struct {
 		name   string
 		args   args
 		fields fields
+		ctx    testContext
 	}{
 		{
-			name: "testing_start_scan",
+			name: "testing_start_scan_no_ctx_timeout",
 			args: args{
 				scanID:     "console",
 				noProgress: true,
@@ -53,14 +62,28 @@ func TestScanner_StartScan(t *testing.T) {
 				types:          []string{""},
 				cloudProviders: []string{""},
 			},
+			ctx: createContext(ctx, time.Second*99999),
+		},
+		{
+			name: "testing_start_scan_with_ctx_timeout",
+			args: args{
+				scanID:     "console",
+				noProgress: true,
+			},
+			fields: fields{
+				types:          []string{""},
+				cloudProviders: []string{""},
+			},
+			ctx: createContext(ctx, time.Second*1),
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			defer tt.ctx.cancel()
 			services, store, err := createServices(tt.fields.types, tt.fields.cloudProviders)
 			require.NoError(t, err)
-			err = StartScan(context.Background(), tt.args.scanID, progress.PbBuilder{}, services)
+			err = StartScan(tt.ctx.ctx, tt.args.scanID, progress.PbBuilder{}, services)
 			require.NoError(t, err)
 			require.NotEmpty(t, &store)
 		})
@@ -130,4 +153,12 @@ func createServices(types, cloudProviders []string) (serviceSlice, *storage.Memo
 		})
 	}
 	return services, store, nil
+}
+
+func createContext(ctx context.Context, timeout time.Duration) testContext {
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	return testContext{
+		ctx,
+		cancel,
+	}
 }

